design/providers/kubernetes_secrets_storage_csi/conjur_csi_provider_solution_design.md

```diff
@@ -0,0 +1,314 @@
+
+By implementing this gRPC server and the `CSIDriverProviderServer` interface, we can create a Conjur provider for the Kubernetes Secrets Store CSI Driver. This provider will handle fetching secrets from the Conjur instance and passing them back to the Secrets Store CSI Driver, which in turn makes them available to Kubernetes workloads.
+
+Helm charts will be created for deploying the CyberArk Conjur provider. This provides a streamlined, automated, and repeatable deployment process.
+
+### UX
+
+The user interaction with our solution will mainly be through Kubernetes manifests and Helm.
+
+#### SecretProviderClass
+
+`SecretProviderClass` is a Kubernetes custom resource that stores the secret provider configuration. The user will define a Kubernetes `SecretProviderClass` object that refers to the CyberArk Conjur provider and specifies necessary parameters for it. This would look something like the following:
+
+```yaml
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: db-credentials
+spec:
+  provider: conjur
+  parameters:
+    authnUrl: https://conjur-auth.some-org.com
+    applianceUrl: https://conjur.some-org.com
+    account: some-org
+    authnLogin: path/to/some-policy/some-host
+    policyPath: path/to/some-policy/with-secrets # defaults to /
+    secrets: |
+      - db_username: "db/username" # relative to policyPath, otherwise if it starts with / then it is absolute
+      - db_password: "db/password"
+```
+
+In this example, the `SecretProviderClass` named `db-credentials` is using the CyberArk Conjur provider. The parameters section contains the `authnUrl`, the `applianceUrl` and a list of secrets to fetch from Conjur. The secrets are identified by their paths in Conjur  (i.e. `db/username` and `db/password`). The `policyPath` parameter can be used to specify the policy branch from which to resolve relative secret paths.
+
+A pod that wants to use these secrets would reference the `SecretProviderClass` in its volumes definition like so:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mypod
+spec:
+  containers:
+  - name: mycontainer
+    image: myimage
+    volumeMounts:
+    - name: secrets-store-inline
+      mountPath: "/mnt/secrets-store"
+      readOnly: true
+  volumes:
+  - name: secrets-store-inline
+    csi:
+      driver: secrets-store.csi.k8s.io
+      readOnly: true
+      volumeAttributes:
+        secretProviderClass: "db-credentials"
+```
+
+In this example, the Pod `mypod` mounts a volume at `/mnt/secrets-store`. This volume is provided by the Secrets Store CSI driver and is configured to fetch secrets as defined by the `db-credentials` SecretProviderClass. The secrets fetched by the driver will be written to files in this volume. For instance, the secret `db/username` will be written to the file `/mnt/secrets-store/db_username` in the pod's filesystem.
```

Comment:
> ...the secret `db/username` will be written to the file `/mnt/secrets-store/db_username` in the pod's filesystem.

Is this default behavior, or target behavior?

Reply:
Default behavior I believe